kateryna/databases.py

```diff
@@ -0,0 +1,140 @@
+    def __init__(self, message):
+        self.message = message
+
+
+class PostgreSQLConnectionRefusedError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+class MySQLConnectionRefusedError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+class Connection:
+    def __init__(self, db_name):
+        self.open = False
+        self.closed = False
+        self.db_name = db_name
+
+    def __str__(self):
+        return "Connection to {} is open:{} close:{}".format(self.db_name, self.open, self.closed)
+
+
+class BaseFramework:
+    allowed_keys = ["key1", "key2", "key3", "key4", "key5", "key6"]
+    allowed_commands = ["CREATE", "INSERT", "DELETE", "SELECT"]
+
+    def __init__(self, **kwargs):
+        if kwargs:
+            for attr, value in kwargs.items():
+                setattr(self, attr, value)
+
+    def connect(self, db_name=None, key=None):
+        if db_name is not None:
+            conn = Connection(db_name)
+            conn.open = True
+            return conn
+
+    def execute(self, command):
+        if command in self.allowed_commands:
+            return command
+        else:
+            raise TypeError("Provide valid command")
+
+    def close(self):
+        raise NotImplementedError
+
+
+class SQLiteApp(BaseFramework):
+    conn = None
+
+    def connect(self, db_name=None, key=None):
+        if hasattr(self, "db_name") and key in self.allowed_keys:
+            self.conn = super(SQLiteApp, self).connect(self.db_name)
+            return self.conn
+        raise SQLiteConnectionRefusedError(
+                    "Please provide valid key or database name to have access")
+
+    def execute(self, command):
+        valid_command = super(SQLiteApp, self).execute(command)
+        return "SQLiteApp. {} command has been executed".format(valid_command)
+
+    def close(self):
+        conn = self.conn
+        conn.open = False
+        conn.closed = True
+        return conn
+
+
+class MySQLApp(BaseFramework):
+
+    def __init__(self, db, **kwargs):
+        self.db = db
+        super(MySQLApp, self).__init__(**kwargs)
+
+    #completely overrides base method
+    def connect(self, db_name=None, key=None):
+        if not self.db.get("data"):
+            raise MySQLConnectionRefusedError("The database is empty")
+        return "Connected to database {}".format(self.db.get("db_name", "Provide name of the database"))
+
+    def execute(self, command):
+        valid_command = super(MySQLApp, self).execute(command)
+        return "MySQLApp. {} command has been executed".format(valid_command)
+
+    def close(self):
+        return "MySQLApp. Connection to {} closed".format(self.db)
+
+
+class PostgreSQLApp(BaseFramework):
+    # override base allowed keys
+    allowed_keys = ["key6", "key7", "key8", "key9", "key10", "key11"]
+
+    def connect(self, db_name=None, key=None):
+        if key in self.allowed_keys:
```

Comment:
I thought try except block is for catching errors but for raising errors we should do smth like that: def functionName( level ):
   if level < 1:
      raise "Invalid level!", level
Isn't that right?